### PR TITLE
fix: various issues with propose cli

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

* fix: issue with propose execution, sender is required for initial proposal but doesnt mean we want to execute
* gas price was not settable
* account state was lost - i think only really matter with MockClient - but fixed
* missing success log when proposing
* filled in some tests surrounding the propose command but benefits i think all command
* handle some 0.7.1 upgrades

coing up next, ill look at approve, add tests, and fill in any gaps there.

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -17,6 +17,16 @@
 ape.config.DATA_FOLDER = DATA_FOLDER
 
 
+@pytest.fixture(scope="session")
+def accounts():
+    return ape.accounts.test_accounts
+
+
+@pytest.fixture(scope="session")
+def chain():
```
Comment (fubuloubu) [inline tests/conftest.py:26]: is this not a session scoped module? or are we not running via `ape test`?

File: ape_safe/accounts.py [hunk 2]
```diff
@@ -56,7 +58,13 @@ def addresses(self) -> Iterator[str]:
...
     def accounts(self) -> Iterator[AccountAPI]:
         for account_file in self._account_files:
-            yield SafeAccount(account_file_path=account_file)
+            if account_file.stem in self._accounts:
+                yield self._accounts[account_file.stem]
+
+            else:
+                acct = SafeAccount(account_file_path=account_file)
+                self._accounts[account_file.stem] = acct
+                yield acct
```
Comment (fubuloubu) [inline ape_safe/accounts.py:67]: this is caching? can add a comment

**Review by fubuloubu — APPROVED** [review]

File: tests/conftest.py (hunk 1 above)
Comment (antazoey) [reply tests/conftest.py:26]: I am not used to plugins running the `ape_test` pytest plugin; i didnt know that would work actually.

Comment (fubuloubu) [reply tests/conftest.py]: it should work just fine yeah haha, the plugin gets loaded naturally if it's installed, even locally installed

**Pull request merged**